src/Components/GenericExpressCheckout/CustomerRegistrationUtil.php

```diff
@@ -145,4 +157,23 @@ private function getCountryIdByCode(string $code, Context $context): ?string
 
         return $country->getId();
     }
+
+    private function hasAddressRequiredData(array $address): bool
+    {
+        $requiredFields = [
```

Comment:
Ah yes, you can make phone, birthday, additionalAddress1 and additionalAddress2 mandatory fields via the configuration. They are certainly not all transferred by PayPal in every case. 

Should I then query the configurations here and, depending on the configuration, make them mandatory fields and if they are not passed, the exception is also thrown, as with the other fields?

Reply:
would be better :)

Reply:
I now query the configurations and treat the other fields as mandatory fields accordingly.

Reply:
Just a thought:

Would it make sense to use this factory?
Shopware\Core\Checkout\Customer\Validation\AddressValidationFactory

So we must not use the awful long keys, which may change in the future. With this we are much more up/backwards compatible, and can just if a not-blank constraint is given :)

If you do not think so, just close the conversion.

Reply:
I like it, thanks for the suggestion! I've implemented it here: https://github.com/PAYONE-GmbH/shopware-6/pull/331/commits/7bd53750070a084775945a03cd254172e9bd419e

Reply:
awesome!

But you should not log objects, because this could cause very large logfiles, just for validation issues:
```
                'billingAddressViolations' => $billingAddressViolations,
                'shippingAddressViolations' => $shippingAddressViolations,
```
You may log only the missing field names.
If you prefer another solution, gor for it :)

Reply:
I now log it with the “__toString()” function of the Violation objects. This then produces a string like this:
![image](https://github.com/user-attachments/assets/fe2f5006-b98a-41f2-b581-7310c49087a0)